File: ruby/code_1a.rb
loc_a = []
loc_b = []

IO.foreach(ARGV[0]) do |line|
    a, b = line.split(' ')
    loc_a.append(a.to_i)
    loc_b.append(b.to_i)
end

loc_a.sort!
loc_b.sort!

res = (0...loc_a.count).sum do |i|
    (loc_a[i] - loc_b[i]).abs
end

puts res

File: java/Code1A.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.util.*;
import java.util.stream.*;

public class Code1A {
    public static void main(String[] args) {
        List<String> file;
        try (
            BufferedReader in = new BufferedReader(new FileReader(args[0]))
        ) {
            file = in.lines().collect(Collectors.toList());
        } catch (Exception e) {
            return;
        }

        List<Integer> locA = new ArrayList<>();
        List<Integer> locB = new ArrayList<>();
        for (String line : file) {
            locA.add(Integer.parseInt(line.split("\s+")[0]));
            locB.add(Integer.parseInt(line.split("\s+")[1]));
        }
        
        locA.sort(null);
        locB.sort(null);

        int res = IntStream.range(0, locA.size()).map(i ->
            Math.abs(locA.get(i) - locB.get(i))
        ).reduce(0, (a,b) -> a+b);

        System.out.println(res);
    }
}